code/modules/antagonists/heretic/knowledge/void_lore.dm

```diff
@@ -275,3 +316,33 @@
 	if(storm)
 		storm.end()
 		QDEL_NULL(storm)
+
+///Few checks to determine if we can deflect bullets
+/datum/heretic_knowledge/ultimate/void_final/proc/can_deflect(mob/living/ascended_heretic)
+	if(!ascended_heretic.combat_mode)
+		return FALSE
+	if(ascended_heretic.incapacitated(IGNORE_GRAB)) //NO STUN
+		return FALSE
+	if(!(ascended_heretic.mobility_flags & MOBILITY_USE)) //NO UNABLE TO USE
+		return FALSE
+	var/datum/dna/dna = ascended_heretic.has_dna()
+	if(dna?.check_mutation(/datum/mutation/human/hulk)) //NO HULK
+		return FALSE
+	if(!isturf(ascended_heretic.loc)) //NO MOTHERFLIPPIN MECHS!
+		return FALSE
+	return TRUE
+
+/datum/heretic_knowledge/ultimate/void_final/proc/hit_by_projectile(mob/living/ascended_heretic, obj/projectile/hitting_projectile, def_zone)
+	SIGNAL_HANDLER
+
+	if(!can_deflect(ascended_heretic))
+		return NONE
+
+	ascended_heretic.visible_message(
+		span_danger("[ascended_heretic] effortlessly swats [hitting_projectile] aside! [ascended_heretic.p_They()] can block bullets with [ascended_heretic.p_their()] bare hands!"),
+		span_userdanger("You deflect [hitting_projectile]!"),
+	)
+	playsound(ascended_heretic, pick('sound/weapons/bulletflyby.ogg', 'sound/weapons/bulletflyby2.ogg', 'sound/weapons/bulletflyby3.ogg'), 75, TRUE)
```

Comment:
This needs better theming than just copying what scarp does, when it was pitched to me I imagined the void bending or something to redirect it back at firer

So like, change the SFX maybe + the message + add a probability that it redirects directly back to the projectile firer instead of random direction

Reply:
Changed the messages and added the probability